### PR TITLE
Feature/dex 859 release config v3

### PR DESCRIPTION
**Pull request by EvansPie**
DEX-859

# What this PR does

Consume config v3 and deprecate properties in `PrimerSettings`.

# Notable decisions & other stuff

List & explain ...

# Instructions on how to test this

Let other reviewers know how they can test this.

# Before merging

_QA_

- [x] I manually tested this with a demo application (simulator)
- [ ] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Data Models/OrderItem.swift [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import Foundation
 import PassKit
 
+@available(*, deprecated, message: "Set the order items in the client session")
```
Comment (xevious78) [inline Sources/PrimerSDK/Classes/Data Models/OrderItem.swift:13]: Not sure it's clear that the client session is something on the backend. Developers would first look at a SDK settings called "client session".

Maybe something like this?
`Set the order items in the client session with POST /client-session`

File: Sources/PrimerSDK/Classes/Data Models/PrimerSettings.swift [hunk 2]
```diff
@@ -6,12 +6,16 @@ public typealias TokenizationSuccessCallBack = (_ paymentMethodToken: PaymentMet
 public typealias CheckoutDismissalCallback = () -> Void
 
 internal protocol PrimerSettingsProtocol {
+    @available(*, deprecated, message: "Use client session")
```
Comment (xevious78) [inline Sources/PrimerSDK/Classes/Data Models/PrimerSettings.swift:9]: Again "Use client session" doesn't help much. Could you rephrase?
"Pass this in the client session with POST /client-session"

**Review by xevious78 — APPROVED** [review]
could you improve the deprecation messages? :)

File: Sources/PrimerSDK/Classes/Data Models/OrderItem.swift (hunk 1 above)
Comment (ceriksson) [reply Sources/PrimerSDK/Classes/Data Models/OrderItem.swift:13]: We could also add a url link to the relevant part in the API documentation.

File: Sources/PrimerSDK/Classes/Data Models/Throwable.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+//  PrimerSDK
+//
+//  Created by Evangelos on 22/11/21.
+//
+
+import Foundation
+
+#if canImport(UIKit)
+
+enum Throwable<T: Decodable>: Decodable {
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Data Models/Throwable.swift:12]: nice!

File: Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/KlarnaTokenizationViewModel.swift [hunk 4]
```diff
@@ -147,13 +147,13 @@ class KlarnaTokenizationViewModel: PaymentMethodTokenizationViewModel, ExternalP
...
             
-            if settings.orderItems.isEmpty {
+            if (settings.orderItems ?? []).isEmpty {
                 let err = KlarnaException.missingOrderItems
                 _ = ErrorHandler.shared.handle(error: err)
                 throw err
             }
             
-            if !settings.orderItems.filter({ $0.unitAmount == nil }).isEmpty {
+            if !(settings.orderItems ?? []).filter({ $0.unitAmount == nil }).isEmpty {
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/KlarnaTokenizationViewModel.swift:156]: maybe the orderItems should default to an empty array so that we don't need to null check each time?

**Review by ceriksson — COMMENTED** [review]
looks great, just the comments to address :)

File: Sources/PrimerSDK/Classes/Data Models/OrderItem.swift (hunk 1 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Data Models/OrderItem.swift:13]: Makes sense. I'll include the URL as well.

File: Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/KlarnaTokenizationViewModel.swift (hunk 4 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/KlarnaTokenizationViewModel.swift:156]: I did that on purpose. This way we know that they haven't been set as long as they're nil, or that they have been set to an empty array (which means we have a response).
